theme-contract/Module.mustache

```diff
@@ -1,32 +1,102 @@
+
+### 3. Dimension Contracts
+Defines abstract sizes, spaces, and borders.
+- **Interfaces:** `OudsBorders`, `OudsSizes`, `OudsSpaces`.
+- **Example:** Defines that a `spaces.fixed.medium` token exists.
+
+### 4. Component Tokens Contract
+Defines semantic tokens specifically scoped to OUDS custom components.
+- **Interface:** `OudsComponentTokens`
+- **Purpose:** Allows overriding default values for specific components without altering the global semantic theme.
+- **Example:** Defines specific tokens for an `OudsButton` (e.g., specific background color) independent of the generic theme.
+
+### 5. Other Contracts
+Includes interfaces for `OudsElevations`, `OudsGrids`, and `OudsOpacities`.
+
+## Usage
+
+### For Application Developers
+**You typically do not interact with this module directly.**
+You will use the *implementation* of this contract provided by the `core` module (via `OudsTheme`). However, understanding this contract helps you know which semantic tokens are guaranteed to exist.
+
+### For Theme Creators
+If you are creating a custom theme implementation (e.g., for a specific sub-brand) while maintaining full compatibility with OUDS components, you must implement `OudsThemeContract`.
+
+## Relationship with other modules
+
+| Module | Role | Analogy |
+| :--- | :--- | :--- |
+| **global-raw-tokens** | The Ingredients | "Flour", "Eggs", "Milk" |
+| **theme-contract** | **The Recipe Template** | **"A cake must have a base, a filling, and icing"** |
+| **core** | The Implementation | "This specific cake uses flour for the base and chocolate for the filling" |
+
+## Tokens versions
 
 <table>
 <tr><td>Android system</td><td>{{AndroidSystem}}</td></tr>
 <tr><td>Orange brand</td><td>{{OrangeBrand}}</td></tr>
 </table>
 
+> 🤖 **Note:** Files in `tokens` packages are **automatically generated by Tokenator** based on the design definitions.
+>
+> Any manual changes to these files will be lost during the next synchronization.
+
 # Package com.orange.ouds.theme
 
-Contains the `OudsThemeContract` interface that can be implemented to create a new theme by overriding semantic and components tokens.
-Other files contains helpers to manage themes.
+This is the root package of the contract module. It primarily defines the `OudsThemeContract` interface.
+This interface is the central point of the design system's abstraction. It aggregates all the specific sub-contracts (colors, typography, borders, spaces, etc.) into a single object that represents a complete theme. Any class claiming to be an OUDS Theme (like `OrangeTheme`) must implement this contract.
 
 # Package com.orange.ouds.theme.tokens
 
-Contains key token files which are tokens identifiers that should be used in your app to get a token value.
-These files are automatically generated by Tokenator.
+This package contains the **Key Token objects** that define the organization of semantic tokens. It establishes the **hierarchy** (tree structure) used to access token values.
```

Comment:
What do you think about moving the key tokens into `com.orange.ouds.theme.tokens.key` or something similar?